klippy/extras/tmc2209.py

```diff
@@ -0,0 +1,92 @@
+# TMC2209 configuration
+#
+# Copyright (C) 2019  Stephan Oelze stephan.oelze@gmail.com
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import tmc2208, tmc2130, tmc, tmc_uart
+TMC_FREQUENCY=12000000.
+
+Registers = dict(tmc2208.Registers)
+
+Registers.update({
+    "TCOOLTHRS": 0x14,
+    "COOLCONF": 0x42,
+    "SGTHRS": 0x40
+})
+
+ReadRegisters = tmc2208.ReadRegisters
+
+fields = dict(tmc2208.Fields)
+fields["COOLCONF"] = {
+    "semin":                    0x0F << 0,
+    "seup":                     0x03 << 5,
+    "semax":                    0x0F << 8,
+    "sedn":                     0x03 << 13,
+    "seimin":                   0x01 << 15
+}
+fields["SGTHRS"] = {
+    "sgt": 0xFF << 0
```

Comment:
The tmc2209 does not have a field named "sgt".  So, I think it's confusing to define one.  The field is SGTHRS on the tmc2209 and it has different meaning than sgt.

Reply:
thats right, changing to
```python
.....
fields["SGTHRS"] = {
    "SGTHRS":      0xFF << 0
}
......
         set_config_field(config, "SGTHRS", 0)
```